### ISSUE TITLE
Init script fails

### ISSUE THREAD

**Issue by GregoryLand** [eoli3n/archiso-zfs#17]
Archlinux ISO Version: archlinux-2024.06.01-x86_64.iso
Install Medium: USB Flash drive

Attempting to run
`$ curl -s https://raw.githubusercontent.com/eoli3n/archiso-zfs/master/init | bash -s -- -v`
results in the output.
`No ZFS module found`

Debug log attached.
[debug.log](https://github.com/user-attachments/files/15543580/debug.log)

**Comment by eoli3n** [comment]
Could you try this please and confirm that dkms fallback is working ?
Note than I pushed on the ``dev`` branch, so you need to replace ``master`` in the url as follow

```
$ curl -s https://raw.githubusercontent.com/eoli3n/archiso-zfs/dev/init | bash -s -- -v
```

**Comment by GregoryLand** [comment]
Test ran.  Log uploaded.  I was able to see existing pool on m.2 drive.

[debug.log](https://github.com/user-attachments/files/15551343/debug.log)

I installed on the machine using 2024.05.01-x86_64.iso as a workaround.  Totally okay to help troubleshoot the issue though.

**Comment by Asday** [comment]
Tje DKMS fallback works on `dev` for me on `archiso-2024.06.01-x86-64.iso` while `master` does not.

**Comment by eoli3n** [comment]
Yes, that's what I fixed, let's merge.
The latest debug.log of @GregoryLand shows that dkms fallback starts, but it fails at some point.
If using previous iso works for now, or/and if DKMS works for @Asday, as I have not much time for this right now, I keep this open and we'll check if the script is still buggy with next month iso.

**Comment by Asday** [comment]
DKMS fallback worked fine for me.  I did get an amusing "it didn't work at all" message followed immediately by "it worked perfectly" message at the end of the script, which I presume was just some output not quite being redirected.  Didn't bother me though, just made me smile.

**Comment by r-maerz** [comment]
If anyone is still struggling getting @eoli3n's script to work, please note that it depends on the archzfs project. The maintainer of said project however disappeared earlier this year, leavin the archzfs project in a difficult position and us users without updates of the zfs-related packages.

I have hence created an archiso that includes both the LTS kernel as well as the openzfs modules for it. My iso will be updated automatically on the 4th day of every month. You may find using my iso more convenient than dealing with all this mess.

It's available on [GitHub](https://github.com/r-maerz/archlinux-lts-zfs) for anyone to use.

**Comment by eoli3n** [comment]
> The maintainer of said project however disappeared earlier this year

Latest commit is last month, I wonder who you are talking about, and why do you state this ?

From your README

> That method however fails when [...] archiso-zfs are not updated in time to address any of the above

How could I fix latest kernel compatibility with ZFS or deprecation or archzfs ? That statement is strange or badly expressed.
Have you tested archiso-zfs with the current iso  ?

**Comment by GregoryLand** [comment]
He is correct.  The community has taken over the repo and working on
correcting the current issues.  The last commit was a series of changes to
allow builds on GitHub because the infrastructure is broken and can't be
fixed since no one currently active in the project has access to it. Just
gonna take time for everyone to get things sorted out.

**Comment by eoli3n** [comment]
I don't say no, but it doesn't answer my question

> I wonder who you are talking about

In my memories, I was dealing with @minextu

**Comment by r-maerz** [comment]
> I don't say no, but it doesn't answer my question
> 
> > I wonder who you are talking about
> 
> In my memories, I was dealing with @minextu
> 
> You could add your project to the wiki `1.3 Download a pre-built archiso with ZFS` to https://wiki.archlinux.org/title/Install_Arch_Linux_on_ZFS

You are correct that you were dealing with minextu. The same minextu who has been unresponsive on any queries related to the archzfs project for over 6 months now. Have a look at the discussion in the archzfs repo to learn all the about that.

English isn't my first language, but you are reading way too much into what I said regarding updates and archiso-zfs. I did indeed try your script at the beginning of this month, because I have been using it happily for over a year now. Precompiled packages from archzfs could not be used at the time as those were out of date and the dkms install failed with an unspecific error message.

**Comment by eoli3n** [comment]
> English isn't my first language, but you are reading way too much into what I said regarding updates and archiso-zfs

I read what is written, so if it is not what you want to say, maybe you could fix it ?
